nupic/research/frameworks/vernon/experiments/self_supervised_experiment.py

```diff
@@ -377,10 +321,20 @@ def create_unsupervised_sampler(cls, config, dataset):
 
     @classmethod
     def create_supervised_sampler(cls, config, dataset):
+        num_samples = config.get("num_supervised_samples", -1)
+        if num_samples > 0:
+            return RandomSampler(dataset, num_samples=num_samples)
+        return None
+
+    @classmethod
+    def create_validation_sampler(cls, config, dataset):
+        num_samples = config.get("num_validation_samples", -1)
+        if num_samples > 0:
+            return RandomSampler(dataset, num_samples=num_samples)
         return None
 
     @classmethod
-    def load_dataset(cls, config, dataset_type="unsupervised"):
+    def load_dataset(cls, config, train=True, dataset_type="unsupervised"):
```

Comment:
You should handle the `train` parameters in the method.
```
if not train:
    dataset_type = "validation"
```

Reply:
added